Build failure: python313Packages.pyhumps

### Nixpkgs version

- Unstable (26.05)

### Steps to reproduce

nix build github:nixos/nixpkgs/nixos-unstable-small#python313Packages.pyhumps

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/322426463

### Relevant log output

```console
=========================== short test summary info ============================
FAILED tests/test_decamelize.py::test_decamelize[_APIResponse_-_api_response_] - AssertionError: __api_response_ != _api_response_
FAILED tests/test_decamelize.py::test_decamelize[__APIResponse__-__api_response__] - AssertionError: ___api_response__ != __api_response__
FAILED tests/test_decamelize.py::test_decamelize_dict_list - AssertionError: [{'symbol': 'AAL', 'last_price': 31.78, 'change_pct': 2.814...
FAILED tests/test_humps.py::test_camelized_acronyms[_HTTPResponse-_http_response] - AssertionError: assert '__http_response' == '_http_response'
FAILED tests/test_humps.py::test_camelized_acronyms[_HTTPResponse__-_http_response__] - AssertionError: assert '__http_response__' == '_http_response__'
FAILED tests/test_humps.py::test_camelized_acronyms[B52Thing-b52_thing] - AssertionError: assert 'b52thing' == 'b52_thing'
FAILED tests/test_humps.py::test_depascalize - AssertionError: assert [{'change_pct...cronym', ...}] == [{'change_pct...cr...
FAILED tests/test_kebabize.py::test_kebabize_dict_list - AssertionError: assert {'images': [{...a.io/video'}]} == {'images': [{...a....
FAILED tests/test_separate_words.py::test_separate_words[_HelloWorld-_Hello_World] - AssertionError: __Hello_World != _Hello_World
FAILED tests/test_separate_words.py::test_separate_words[__HelloWorld-__Hello_World] - AssertionError: ___Hello_World != __Hello_World
======================== 10 failed, 107 passed in 0.12s ========================
```

### Additional context

This was discovered when trying to build Mealie on `nixos-unstable-small`

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.18.7, NixOS, 26.05 (Yarara), 26.05.20260123.c5296fd`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Determinate Nix 3.15.2) 2.33.1`
 - nixpkgs: `/nix/store/8hx8jwa24q6rkzhca9iqy5ckk9rbphi1-source`

### Notify maintainers

@fabaff

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+label%3A%220.kind%3A+build+failure%22).
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/python-modules/pyhumps/default.nix
+++ b/pkgs/development/python-modules/pyhumps/default.nix
@@ -4,12 +4,22 @@
   fetchFromGitHub,
   poetry-core,
   pytestCheckHook,
+  fetchpatch2,
 }:
 
 buildPythonPackage rec {
   pname = "pyhumps";
   version = "3.9.0";
   pyproject = true;
+
+  patches = [
+    (fetchpatch2 {
+      name = "revert_decamelize_ignore_numeric_characters.patch";
+      revert = true;
+      url = "https://github.com/nficano/humps/commit/661828b8b3e6fac15236b14ffbdf038aef516d4c.patch";
+      hash = "sha256-5+ZKEqydN+SF2ihWUf7B8TBY2DK7xFIU/WRDS3+pD0k=";
+    })
+  ];
 
   src = fetchFromGitHub {
     owner = "nficano";

